eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,194 @@
+            final int childType = child.getType();
+
+            if (childType == JavadocTokenTypes.NEWLINE || childType == JavadocTokenTypes.LEADING_ASTERISK) {
+                log(child.getLineNumber(), child.getColumnNumber(), "No multiple lines in @code annotation");
+            }
+
+            // This code section duplicates the checking on @codesnippet, maven build already failed it
+            if (childType == JavadocTokenTypes.DESCRIPTION
+                && (!JavadocUtil.containsInBranch(child, JavadocTokenTypes.NEWLINE)
+                    || !JavadocUtil.containsInBranch(child, JavadocTokenTypes.LEADING_ASTERISK))) {
+                log(child.getLineNumber(), child.getColumnNumber(), "No multiple lines in @codesnippet annotation");
+            }
+        }
+        return;
+    }
+
+    /**
+     * Check to see if the code snippet name matched the naming pattern.
+     *
+     * @param inlineTagNode JAVADOC_INLINE_TAG javadoc node
+     */
+    private void checkCodeSnippetNaming(DetailNode inlineTagNode) {
+        if (!isCodeSnippet(inlineTagNode)) {
+            return;
+        }
+
+        final DetailNode descriptionNode = JavadocUtil.findFirstToken(inlineTagNode, JavadocTokenTypes.DESCRIPTION);
+        if (descriptionNode == null) {
+            return;
+        }
+
+        final DetailNode textNode = JavadocUtil.findFirstToken(descriptionNode, JavadocTokenTypes.TEXT);
+        if (textNode == null) {
+            return;
+        }
+
+        final String namingPattern = textNode.getText();
+        // Verify naming pattern spec:
+        // 1, For package name, class name, they should all be lower case;
+        //    For method name,  the first letter of method name should be lower case.
+        //    For parameters, they should only be letters. Concatenate all parameters with dash character
+        // 2, Jonathon will add more requirement specs later
+        if (!isNamingMatch(namingPattern)) {
+            log(textNode.getLineNumber(), textNode.getColumnNumber(),
+                String.format("Naming pattern mismatch. It should only contain lower case for the package path and class name for matching regular expression ''%s''" +
+                    " and the first letter of method name must be small case for matching regular expression ''%s'', and the parameters should be all small case for matching " +
+                    "regular expression ''%s''.", CLASS_PATH_REGEX, METHOD_NAME_REGEX, PARAMETERS_REGEX));
+        }
+    }
+
+    /**
+     * Find if the given JAVADOC_INLINE_TAG is a @codesnippet tag.
+     *
+     * @param inlineTagNode JAVADOC_INLINE_TAG javadoc node
+     * @return true if it is a '@codesnippet' tag, false otherwise.
+     */
+    private boolean isCodeSnippet(DetailNode inlineTagNode) {
+        final DetailNode customNameNode = JavadocUtil.findFirstToken(inlineTagNode, JavadocTokenTypes.CUSTOM_NAME);
+        if (customNameNode == null) {
```

Comment:
this can be simplified: `return customNameNode != null && customNameNode.getText().equals("@codesnippet")`